### PR TITLE
HAproxy should read config path from confstore.

### PR DESCRIPTION
**Pull request by shahidSTX**
# Problem Statement
- HAproxy should read config path from confstore, remove hard-coding.

# Design
-  HAproxy should read config path from confstore, remove hard-coding.

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: scripts/provisioning/s3_haproxy_config.py [hunk 1]
```diff
@@ -358,11 +368,12 @@ def configure_haproxy_k8(self):
...
         os.makedirs(pem_dir)
-    if not os.path.exists('/etc/cortx/haproxy/errors/'):
-        os.makedirs('/etc/cortx/haproxy/errors/')
+    errors_path = os.path.join(baseconfig_path, 'haproxy/errors/')
+    if not os.path.exists(errors_path):
+        os.makedirs(errors_path)
 
     #Run config commands
-    os.system("cp /opt/seagate/cortx/s3/install/haproxy/503.http /etc/cortx/haproxy/errors/")
+    shutil.copy('/opt/seagate/cortx/s3/install/haproxy/503.http', errors_path)
```
Comment (t7ko-seagate) [inline scripts/provisioning/s3_haproxy_config.py:376]: use this correct path in the config itself, line 271

```
errorfile               503 /etc/haproxy/errors/503.http
```

File: scripts/provisioning/s3_haproxy_config.py [hunk 2]
```diff
@@ -358,11 +368,12 @@ def configure_haproxy_k8(self):
     pem_dir = os.path.dirname(sslcertpath)
```
Comment (t7ko-seagate) [inline scripts/provisioning/s3_haproxy_config.py:368]: Need to use correct ssl cert path in haproxy config above, line 361.  Not sure if this cert is the same as sslcertpath or not.

```
server s3authserver-instance1 0.0.0.0:%s #check ssl verify required ca-file /etc/ssl/stx-s3/s3auth/s3authserver.crt   
```

**Review by t7ko-seagate — CHANGES_REQUESTED** [review]

File: scripts/provisioning/s3_haproxy_config.py (hunk 2 above)
Comment (shahidSTX) [reply scripts/provisioning/s3_haproxy_config.py:368]: In line 361 - `server s3authserver-instance1 0.0.0.0:%s` we are adding port number, not ssl certificate path.

File: scripts/provisioning/s3_haproxy_config.py (hunk 1 above)
Comment (shahidSTX) [reply scripts/provisioning/s3_haproxy_config.py:376]: Done.

File: scripts/provisioning/s3_haproxy_config.py [hunk 3]
```diff
@@ -196,7 +206,9 @@ def configure_haproxy_k8(self):
     s3backendport = self.get_s3server_backend_port()
     s3authbendport = self.get_s3auth_backend_port()
 
-    config_file = '/etc/cortx/s3/haproxy.cfg'
+    config_file = os.path.join(baseconfig_path, 's3/haproxy.cfg')
+    errors_path = os.path.join(baseconfig_path, 'haproxy/errors/')
```
Comment (nileshgovande) [inline scripts/provisioning/s3_haproxy_config.py:210]: Errors file should be - /opt/seagate/cortx/s3/install/haproxy/503.http - static

**Review by t7ko-seagate — APPROVED** [review]

**Pull request merged**